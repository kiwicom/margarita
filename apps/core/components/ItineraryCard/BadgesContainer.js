// @flow

import * as React from 'react';
import { ScrollView } from 'react-native';
import { Badge, StyleSheet } from '@kiwicom/universal-components';

type BadgeWithId = { ...React.ElementProps<typeof Badge>, id: string };

type Props = {|
  +badges?: Array<BadgeWithId>,
|};

export default function BadgesContainer({ badges }: Props) {
  if (badges == null) {
    return null;
  }
  return (
    <ScrollView horizontal style={styles.container}>
      <>
        {badges.map(badge => (
          <Badge key={badge.id} type={badge.type} style={styles.badge}>
            {badge.children}
          </Badge>
        ))}
      </>
    </ScrollView>
  );
}
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  badge: {
    marginEnd: 10,
  },
});
