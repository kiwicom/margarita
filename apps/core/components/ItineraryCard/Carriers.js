// @flow

import * as React from 'react';
import { Platform } from 'react-native';
import { CarrierLogo } from '@kiwicom/universal-components';
import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
import { uniq } from 'ramda';

import type { Carriers_data as CarriersType } from './__generated__/Carriers_data.graphql';

type SegmentsType = $PropertyType<CarriersType, 'segments'>;

type Props = {|
  +data: ?CarriersType,
|};

const mapCarriers = (segments: SegmentsType) => {
  const carriers =
    segments &&
    uniq(
      segments.map(segment => ({
        name: segment?.carrier?.name,
        code: segment?.carrier?.code,
      })),
    );

  return carriers;
};

function Carriers({ data }: Props) {
  const carriers = mapCarriers(data?.segments);

  if (carriers == null) {
    return null;
  }

  return (
    <CarrierLogo
      size={Platform.OS === 'web' ? 'large' : 'medium'}
      carriers={carriers}
    />
  );
}

export default createFragmentContainer(Carriers, {
  data: graphql`
    fragment Carriers_data on Sector {
      segments {
        carrier {
          name
          code
        }
      }
    }
  `,
});
