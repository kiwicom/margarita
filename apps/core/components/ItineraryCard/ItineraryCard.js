// @flow

import * as React from 'react';
import { FlatList, View } from 'react-native';
import { formatPrice } from '@kiwicom/margarita-utils';
import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
import { TouchableWithoutFeedback } from '@kiwicom/margarita-components';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';
import { StyleSheet, Hoverable, Card } from '@kiwicom/universal-components';

import RenderTripSectorItem from './RenderTripSectorItem';
import ItineraryCardWrapper from './ItineraryCardWrapper';
import ItineraryDetail from './itineraryDetail/ItineraryDetail';
import type { ItineraryCard as ItineraryCardType } from './__generated__/ItineraryCard.graphql';
import type { RenderTripSectorItem as RenderTripSectorItemType } from './__generated__/RenderTripSectorItem.graphql';

type Props = {|
  +data: ItineraryCardType,
  +onBookPress: () => void,
|};

type State = {|
  detailOpened: boolean,
  hovered: boolean,
|};

type SectorItem = {|
  item: ?RenderTripSectorItemType,
|};

class ItineraryCard extends React.Component<Props, State> {
  state = {
    detailOpened: false,
    hovered: false,
  };

  handleOnMouseEnter = () => {
    this.setState({ hovered: true });
  };

  handleOnMouseLeave = () => {
    this.setState({ hovered: false });
  };

  handlePress = () => {
    this.setState(state => ({
      detailOpened: !state.detailOpened,
    }));
  };

  handleDetailClose = () => {
    this.setState({ detailOpened: false });
  };

  renderSectorItem = ({ item }: SectorItem) => {
    if (item) {
      return <RenderTripSectorItem data={item} />;
    }
    return null;
  };

  keyExtractor = (_, index) => `${index}`;

  render() {
    const { detailOpened, hovered } = this.state;
    const { data, onBookPress } = this.props;
    if (data == null) {
      return null;
    }

    const priceObject = {
      amount: parseFloat(data.price?.amount) ?? 0,
      currency: data.price?.currency ?? 'CZK',
    };
    const localizedPrice = formatPrice(priceObject);

    return (
      <Card
        style={[
          styles.container,
          (hovered || detailOpened) && styles.containerElevated,
        ]}
      >
        <Hoverable
          onMouseEnter={this.handleOnMouseEnter}
          onMouseLeave={this.handleOnMouseLeave}
        >
          <TouchableWithoutFeedback onPress={this.handlePress}>
            <View>
              <ItineraryCardWrapper
                localizedPrice={localizedPrice}
                detailOpened={detailOpened}
              >
                <FlatList
                  data={data.sectors}
                  keyExtractor={this.keyExtractor}
                  renderItem={this.renderSectorItem}
                />
              </ItineraryCardWrapper>
            </View>
          </TouchableWithoutFeedback>
        </Hoverable>
        {detailOpened && (
          <ItineraryDetail
            localizedPrice={localizedPrice}
            onClose={this.handleDetailClose}
            onBookPress={onBookPress}
          />
        )}
      </Card>
    );
  }
}

export default createFragmentContainer(
  ItineraryCard,
  graphql`
    fragment ItineraryCard on Itinerary {
      sectors {
        ...RenderTripSectorItem
      }
      price {
        currency
        amount
      }
    }
  `,
);

const styles = StyleSheet.create({
  container: {
    padding: 0,
    ios: {
      marginBottom: 8,
    },
    android: {
      marginVertical: 4,
    },
    web: {
      alignSelf: 'center',
      width: '100%',
      maxWidth: 720,
      marginBottom: parseInt(defaultTokens.spaceMedium, 10),
      borderRadius: parseInt(defaultTokens.borderRadiusNormal, 10),
      boxShadow: '0 2px 4px 0 rgba(23,27,30,.1)',
      transitionDuration: '0.3s',
      transitionProperty: 'box-shadow',
      transitionTimingFunction: 'ease-out',
    },
  },
  containerElevated: {
    web: {
      boxShadow: '0 4px 12px 0 rgba(23,27,30,.3)',
    },
  },
});
