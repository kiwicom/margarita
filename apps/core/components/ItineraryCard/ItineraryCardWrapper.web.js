// @flow

import * as React from 'react';
import { View } from 'react-native';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';
import { margaritaTokens } from '@kiwicom/margarita-design-tokens';
import {
  StyleSheet,
  Icon,
  LocalizedPrice,
  Button,
} from '@kiwicom/universal-components';
import {
  withLayoutContext,
  LAYOUT,
  type LayoutContextState,
  noop,
} from '@kiwicom/margarita-utils';

type Props = {|
  +localizedPrice: string,
  +children: React.Node,
  +detailOpened?: boolean,
  +layout: number,
|};

const ItineraryCardWrapper = ({
  localizedPrice,
  children,
  detailOpened,
  layout,
}: Props) => {
  const mobileLayout = layout < LAYOUT.largeMobile;
  const chevronIconName = detailOpened ? 'chevron-up' : 'chevron-down';

  return (
    <View style={styles.card}>
      <View style={styles.row}>
        {!mobileLayout && (
          <View style={styles.price}>
            <LocalizedPrice localizedPrice={localizedPrice} />
          </View>
        )}
        <View style={styles.sectors}>{children}</View>
        {!mobileLayout && (
          <Icon
            name={chevronIconName}
            color={defaultTokens.paletteInkLighter}
          />
        )}
      </View>
      {mobileLayout && (
        <View style={styles.footer}>
          <LocalizedPrice localizedPrice={localizedPrice} />
          <Button
            label={detailOpened ? 'Hide' : 'Show Detail'}
            type="secondary"
            rightIcon={<Icon name={chevronIconName} />}
            onPress={noop}
          />
        </View>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    padding: parseInt(defaultTokens.spaceMedium, 10),
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  price: {
    minWidth: 80,
    marginEnd: 10,
  },
  sectors: {
    flex: 1,
  },
  footer: {
    marginTop: parseInt(defaultTokens.spaceMedium, 10),
    paddingTop: parseInt(defaultTokens.spaceMedium, 10),
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderStyle: 'dotted',
    borderColor: margaritaTokens.borderColorInkLight,
    borderTopWidth: parseInt(defaultTokens.borderWidthCard, 10),
  },
});

const layoutSelect = ({ layout }: LayoutContextState) => ({
  layout,
});

export default withLayoutContext(layoutSelect)(ItineraryCardWrapper);
