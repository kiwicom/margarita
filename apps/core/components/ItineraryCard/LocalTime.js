// @flow

import * as React from 'react';
import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
import { type StylePropType, Text } from '@kiwicom/universal-components';

import { getFormattedDate } from './TripSectorHelpers';
import type { LocalTime as LocalTimeType } from './__generated__/LocalTime.graphql';

type Props = {|
  +data: ?LocalTimeType,
  +dateFormat?: ?string,
  +style?: StylePropType,
|};

function LocalTime({ data, dateFormat, style }: Props) {
  return (
    <Text style={style} numberOfLines={1}>
      {getFormattedDate(data?.time?.local, dateFormat)}
    </Text>
  );
}

export default createFragmentContainer(
  LocalTime,
  graphql`
    fragment LocalTime on RouteStop {
      time {
        local
      }
    }
  `,
);
