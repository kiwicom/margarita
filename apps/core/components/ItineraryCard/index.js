// @flow

export { default as ItineraryCard } from './ItineraryCard';
