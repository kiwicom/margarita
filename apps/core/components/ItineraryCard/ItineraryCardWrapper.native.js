// @flow

import * as React from 'react';
import { View } from 'react-native';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';
import { StyleSheet, LocalizedPrice } from '@kiwicom/universal-components';

import ItineraryCardRow from './ItineraryCardRow';
import BadgesContainer from './BadgesContainer';
import HorizontalDash from './HorizontalDash';

type Props = {|
  +localizedPrice: string,
  +children: React.Node,
  +detailOpened?: boolean,
|};

export default function ItineraryCardWrapper({
  localizedPrice,
  children,
}: Props) {
  // @TODO use real badges
  const badges = [
    {
      id: '1',
      type: 'warning',
      children: 'Cheapest',
    },
    {
      id: '2',
      type: 'neutral',
      children: 'Wi-fi',
    },
  ];

  return (
    <View style={styles.card}>
      {children}
      <HorizontalDash />
      <ItineraryCardRow style={styles.footer}>
        <BadgesContainer badges={badges} />
        <LocalizedPrice localizedPrice={localizedPrice} />
      </ItineraryCardRow>
    </View>
  );
}

const styles = StyleSheet.create({
  card: {
    borderColor: defaultTokens.borderColorCard,
    borderBottomWidth: parseInt(defaultTokens.borderWidthCard, 10),
    borderTopWidth: parseInt(defaultTokens.borderWidthCard, 10),
  },
  footer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginBottom: 5,
    marginTop: 2,
  },
});
