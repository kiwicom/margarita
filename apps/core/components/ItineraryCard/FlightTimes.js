// @flow

import * as React from 'react';
import { View, Platform } from 'react-native';
import { defaultTokens } from '@kiwicom/orbit-design-tokens';
import { StyleSheet, Text, designTokens } from '@kiwicom/universal-components';
import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';

import type { FlightTimes_data as FlightTimesType } from './__generated__/FlightTimes_data.graphql';
import LocalTime from './LocalTime';

type Props = {|
  +data: ?FlightTimesType,
|};

const FlightTimes = (props: Props) => (
  <View style={styles.container}>
    <LocalTime data={props.data?.departure} style={styles.text} />
    {Platform.OS === 'web' && <Text style={styles.text}> - </Text>}
    <LocalTime data={props.data?.arrival} style={styles.text} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    minWidth: 55,
    web: {
      flexDirection: 'row',
    },
  },
  text: {
    fontWeight: 'bold',
    fontSize: parseFloat(defaultTokens.fontSizeTextNormal),
    color: defaultTokens.colorTextAttention,
    padding: 5,
    web: {
      fontWeight: designTokens.fontWeightMedium,
      color: designTokens.paletteBlueDark,
      padding: 0,
      lineHeight: 17,
    },
  },
});

export default createFragmentContainer(FlightTimes, {
  data: graphql`
    fragment FlightTimes_data on Sector {
      arrival {
        ...LocalTime_data
      }
      departure {
        ...LocalTime_data
      }
    }
  `,
});
