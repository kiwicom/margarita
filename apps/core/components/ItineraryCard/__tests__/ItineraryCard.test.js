// @flow

import * as React from 'react';
import { create } from 'react-test-renderer';
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils';
import { QueryRenderer, graphql } from '@kiwicom/relay';

import ItineraryCard from '../ItineraryCard';

const onResponse = props => {
  return <ItineraryCard data={props.searchData.edges[0].node} />;
};
test('ItineraryCard', () => {
  const environment = createMockEnvironment();
  const TestRenderer = () => (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query ItineraryCardQuery @relay_test_operation {
          searchData: searchOneWayItineraries(
            input: {
              itinerary: {
                origin: { ids: ["abc"] }
                destination: { ids: ["def"] }
                outboundDate: { start: "2019-01-01" }
              }
            }
          ) {
            edges {
              node {
                ...ItineraryCard_data
              }
            }
          }
        }
      `}
      variables={{}}
      onResponse={onResponse}
    />
  );

  const renderer = create(<TestRenderer />);
  environment.mock.resolveMostRecentOperation(operation =>
    MockPayloadGenerator.generate(operation, {
      Price: () => ({
        amount: 12.2,
        currency: 'EUR',
      }),
      RouteStop: () => ({
        time: {
          local: '2019-01-01',
        },
      }),
    }),
  );

  expect(renderer).toMatchSnapshot();
});
