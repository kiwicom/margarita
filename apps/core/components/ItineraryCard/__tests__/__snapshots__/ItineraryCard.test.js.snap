// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryCard 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#e8edf1",
        "borderStyle": "solid",
        "borderTopWidth": 0.5,
        "padding": 10,
      },
      Array [
        Object {
          "marginBottom": 8,
          "padding": 0,
        },
        false,
      ],
    ]
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityTraits="button"
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 15,
              "paddingVertical": 5,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "paddingStart": 103,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "borderRadius": 3,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "paddingHorizontal": 4,
                },
                Object {
                  "backgroundColor": "#f5f7f9",
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "margin": 0,
                  },
                  Object {
                    "fontFamily": "Roboto",
                  },
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0.3,
                      "lineHeight": 18,
                    },
                    Object {},
                    Object {
                      "color": "#171b1e",
                    },
                  ],
                ]
              }
            >
              Stays 
              42 minutes
               in &lt;mock-value-for-field-"name"&gt;
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 15,
              "paddingVertical": 5,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "height": 70,
            }
          }
        >
          <View
            style={
              Object {
                "paddingEnd": 16,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "display": "flex",
                    "flexDirection": "column",
                    "flexWrap": "wrap",
                    "justifyContent": "space-between",
                  },
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                  false,
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://images.kiwi.com/airlines/64/<mock-value-for-field-\\"code\\">.png?default=airline.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 3,
                    },
                    Object {
                      "height": 24,
                      "width": 24,
                    },
                    false,
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  Object {},
                ]
              }
            >
              <View
                style={
                  Object {
                    "minWidth": 55,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "margin": 0,
                      },
                      Object {
                        "fontFamily": "Roboto",
                      },
                      Object {
                        "color": "#171b1e",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "padding": 5,
                      },
                    ]
                  }
                >
                  0:00
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "margin": 0,
                      },
                      Object {
                        "fontFamily": "Roboto",
                      },
                      Object {
                        "color": "#171b1e",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "padding": 5,
                      },
                    ]
                  }
                >
                  0:00
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "minHeight": 35,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#bac7d5",
                      "borderRadius": 50,
                      "height": 6,
                      "width": 6,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#e8edf1",
                        "flex": 1,
                        "width": 2,
                      },
                      Object {
                        "marginBottom": -5,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "marginBottom": 3,
                        "position": "absolute",
                        "top": 0,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "zIndex": 825,
                      },
                      Object {
                        "marginTop": -4,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 7,
                          "position": "relative",
                          "width": 7,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#cad5df",
                            "position": "absolute",
                          },
                          Object {
                            "backgroundColor": "#bac7d5",
                            "borderBottomStartRadius": 3,
                            "borderTopStartRadius": 3,
                            "bottom": 0,
                            "height": 2,
                            "position": "absolute",
                            "width": 7,
                          },
                          Object {
                            "backgroundColor": "#fff",
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#cad5df",
                            "position": "absolute",
                          },
                          Object {
                            "backgroundColor": "#bac7d5",
                            "borderTopEndRadius": 3,
                            "borderTopStartRadius": 3,
                            "end": 0,
                            "height": 7,
                            "position": "absolute",
                            "width": 2,
                          },
                          Object {
                            "backgroundColor": "#fff",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "marginBottom": 3,
                        "position": "absolute",
                        "top": 0,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "zIndex": 825,
                      },
                      Object {
                        "marginTop": 4,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 7,
                          "position": "relative",
                          "width": 7,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#cad5df",
                            "position": "absolute",
                          },
                          Object {
                            "backgroundColor": "#bac7d5",
                            "borderBottomStartRadius": 3,
                            "borderTopStartRadius": 3,
                            "bottom": 0,
                            "height": 2,
                            "position": "absolute",
                            "width": 7,
                          },
                          Object {
                            "backgroundColor": "#fff",
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#cad5df",
                            "position": "absolute",
                          },
                          Object {
                            "backgroundColor": "#bac7d5",
                            "borderTopEndRadius": 3,
                            "borderTopStartRadius": 3,
                            "end": 0,
                            "height": 7,
                            "position": "absolute",
                            "width": 2,
                          },
                          Object {
                            "backgroundColor": "#fff",
                          },
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "marginBottom": 3,
                        "position": "absolute",
                        "top": 0,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "zIndex": 825,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 7,
                          "position": "relative",
                          "width": 7,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#cad5df",
                            "position": "absolute",
                          },
                          Object {
                            "backgroundColor": "#bac7d5",
                            "borderBottomStartRadius": 3,
                            "borderTopStartRadius": 3,
                            "bottom": 0,
                            "height": 2,
                            "position": "absolute",
                            "width": 7,
                          },
                          undefined,
                        ]
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#cad5df",
                            "position": "absolute",
                          },
                          Object {
                            "backgroundColor": "#bac7d5",
                            "borderTopEndRadius": 3,
                            "borderTopStartRadius": 3,
                            "end": 0,
                            "height": 7,
                            "position": "absolute",
                            "width": 2,
                          },
                          undefined,
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#e8edf1",
                        "flex": 1,
                        "width": 2,
                      },
                      Object {
                        "marginTop": -3,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "#bac7d5",
                      "borderRadius": 50,
                      "height": 6,
                      "width": 6,
                    }
                  }
                />
              </View>
              <View
                style={Object {}}
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "margin": 0,
                      },
                      Object {
                        "fontFamily": "Roboto",
                      },
                      Object {
                        "fontSize": 12,
                        "lineHeight": 17,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  &lt;mock-value-for-field-"name"&gt;
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "margin": 0,
                      },
                      Object {
                        "fontFamily": "Roboto",
                      },
                      Object {
                        "fontSize": 12,
                        "lineHeight": 17,
                        "padding": 5,
                      },
                    ]
                  }
                >
                  &lt;mock-value-for-field-"name"&gt;
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-end",
                    "flex": 1,
                  },
                  Object {},
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "margin": 0,
                    },
                    Object {
                      "fontFamily": "Roboto",
                    },
                    Array [
                      Object {
                        "color": "#7f91a8",
                        "fontSize": 12,
                        "lineHeight": 17,
                        "padding": 5,
                      },
                      Object {},
                    ],
                  ]
                }
              >
                Tue 1 Jan
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "margin": 0,
                      },
                      Object {
                        "fontFamily": "Roboto",
                      },
                      Array [
                        Object {
                          "color": "#7f91a8",
                          "fontSize": 12,
                        },
                        Array [
                          Object {
                            "color": "#7f91a8",
                            "fontSize": 12,
                            "lineHeight": 17,
                            "padding": 5,
                          },
                          Object {},
                        ],
                      ],
                    ]
                  }
                >
                  0h 42m
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderStyle": "dotted",
              "height": 0,
              "position": "relative",
              "width": "100%",
            },
            Object {
              "borderColor": "#e8edf1",
              "borderWidth": 2,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "position": "absolute",
                "width": 750,
              },
              Object {
                "backgroundColor": "#fff",
                "bottom": -2,
                "height": 2,
                "left": -2,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 15,
              "paddingVertical": 5,
            },
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "marginBottom": 5,
              "marginTop": 2,
            },
          ]
        }
      >
        <RCTScrollView
          horizontal={true}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View>
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "borderRadius": 3,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "paddingHorizontal": 4,
                  },
                  Object {
                    "backgroundColor": "#F9971E",
                  },
                  Object {
                    "marginEnd": 10,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "margin": 0,
                    },
                    Object {
                      "fontFamily": "Roboto",
                    },
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 12,
                        "fontWeight": "500",
                        "letterSpacing": 0.3,
                        "lineHeight": 18,
                      },
                      Object {},
                      Object {
                        "color": "#FFFFFF",
                      },
                    ],
                  ]
                }
              >
                Cheapest
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "borderRadius": 3,
                    "flexDirection": "row",
                    "flexWrap": "wrap",
                    "paddingHorizontal": 4,
                  },
                  Object {
                    "backgroundColor": "#f5f7f9",
                  },
                  Object {
                    "marginEnd": 10,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "margin": 0,
                    },
                    Object {
                      "fontFamily": "Roboto",
                    },
                    Array [
                      Object {
                        "color": "#fff",
                        "fontSize": 12,
                        "fontWeight": "500",
                        "letterSpacing": 0.3,
                        "lineHeight": 18,
                      },
                      Object {},
                      Object {
                        "color": "#171b1e",
                      },
                    ],
                  ]
                }
              >
                Wi-fi
              </Text>
            </View>
          </View>
        </RCTScrollView>
        <Text
          style={
            Array [
              Object {
                "margin": 0,
              },
              Object {
                "fontFamily": "Roboto",
              },
              Array [
                Object {
                  "color": "#00826f",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
                undefined,
              ],
            ]
          }
        >
          €12.20
        </Text>
      </View>
    </View>
  </View>
</View>
`;
